### PR TITLE
use pyproject.toml and remove versioneer

### PR DESCRIPTION
**Pull request by AurelienJaquier**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+[build-system]
+requires = ["setuptools >= 64", "setuptools-scm>=8.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "bluepyopt"
+authors = [
+    {name = "Blue Brain Project, EPFL", email = "werner.vangeit@epfl.ch"},
```
Comment (ilkilic) [inline pyproject.toml:8]: I am not sure that they can reach him through this email address anymore

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+]
+description="Bluebrain Python Optimisation Library (bluepyopt)"
```
Comment (ilkilic) [inline pyproject.toml:10]: Perhaps a space between blue and brain?

File: pyproject.toml (hunk 1 above)
Comment (AurelienJaquier) [reply pyproject.toml:8]: I think it is still valid because he is an external collaborator

File: pyproject.toml (hunk 2 above)
Comment (AurelienJaquier) [reply pyproject.toml]: Good catch. Fixed in latest commit

**Review by ilkilic — APPROVED** [review]

**Pull request merged**